.ci/docker/buildtools-windows/Dockerfile

```
@@ -6,21 +6,18 @@ FROM mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-1909
 # Restore the default Windows shell for correct batch processing.
 SHELL ["cmd", "/S", "/C"]
 
-# Download the Build Tools bootstrapper.
-ADD https://aka.ms/vs/16/release/vs_buildtools.exe C:\TEMP\vs_buildtools.exe
+# Download NuGet v5.4.0 
+ADD https://dist.nuget.org/win-x86-commandline/v5.4.0/nuget.exe C:\nuget\nuget.exe
 
-# Install Build Tools excluding workloads and components with known issues.
-RUN C:\TEMP\vs_buildtools.exe --quiet --wait --norestart --nocache `
-    --installPath C:\BuildTools `
-    --all `
-    --remove Microsoft.VisualStudio.Component.Windows10SDK.10240 `
-    --remove Microsoft.VisualStudio.Component.Windows10SDK.10586 `
-    --remove Microsoft.VisualStudio.Component.Windows10SDK.14393 `
-    --remove Microsoft.VisualStudio.Component.Windows81SDK `
- || IF "%ERRORLEVEL%"=="3010" EXIT 0
+# Add NuGet to Beginning of Path (We want this version of Nuget before any other versions that may be in the path)
+RUN @"%SystemRoot%\System32\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -InputFormat None -ExecutionPolicy Bypass -Command "[Environment]::SetEnvironmentVariable(\"Path\", \"C:\nuget;\" + $env:Path, [EnvironmentVariableTarget]::Machine)"
 
-# Start developer command prompt with any other commands specified.
-ENTRYPOINT C:\BuildTools\Common7\Tools\VsDevCmd.bat &&
+# We download Visual Studio from Microsoft
+# If this fails see the Readme: https://github.com/elastic/apm-agent-dotnet/tree/master/.ci/docker/buildtools-windows
+ADD https://download.visualstudio.microsoft.com/download/pr/b79d5d47-f792-4eec-adcf-2da2dfa3e3db/4a244b3dd7c258763614d6e4eb914d42f47917e7cc6a70b509e9d9029aceddf2/vs_Enterprise.exe C:\temp\vs_enterprise.exe
+
+# Install Visual Studio Enterprise with Only packages necessary to build apm-agent-dotnet
+RUN C:\temp\vs_enterprise__833835998.1575916908.exe --quiet --wait --norestart --nocache --add Microsoft.Net.Core.Component.SDK.2.1 --add Microsoft.VisualStudio.Component.Web --add Microsoft.Component.MSBuild
```

Cosmetic change regarding the readability with a multi string command:

```suggestion
RUN C:\temp\vs_enterprise__833835998.1575916908.exe `
         --quiet `
         --wait `
         --norestart `
         --nocache `
         --add Microsoft.Net.Core.Component.SDK.2.1 `
         --add Microsoft.VisualStudio.Component.Web `
         --add Microsoft.Component.MSBuild
```

Although it does require to add the shebang # escape=` at the very beginning of the Dockerfile. For instance https://github.com/PowerShell/PowerShell-Docker/blob/master/release/preview/windowsservercore/docker/Dockerfile#L1